src/main/java/uk/ac/ebi/ena/readtools/loader/fastq/IlluminaIterativeEaterIterator.java

```diff
@@ -23,15 +23,15 @@
 
 import uk.ac.ebi.ena.readtools.loader.common.FileCompression;
 import uk.ac.ebi.ena.readtools.loader.common.QualityNormalizer;
-import uk.ac.ebi.ena.readtools.loader.common.eater.DataEater;
-import uk.ac.ebi.ena.readtools.loader.common.eater.DataEaterException;
-import uk.ac.ebi.ena.readtools.loader.common.feeder.AbstractDataFeeder;
-import uk.ac.ebi.ena.readtools.loader.common.feeder.DataFeederException;
+import uk.ac.ebi.ena.readtools.loader.common.eater.DataConsumer;
+import uk.ac.ebi.ena.readtools.loader.common.eater.DataConsumerException;
+import uk.ac.ebi.ena.readtools.loader.common.feeder.AbstractDataProducer;
+import uk.ac.ebi.ena.readtools.loader.common.feeder.DataProducerException;
 import uk.ac.ebi.ena.readtools.loader.fastq.DataSpot.DataSpotParams;
 import uk.ac.ebi.ena.readtools.loader.fastq.IlluminaIterativeEater.READ_TYPE;
 
 public class 
-IlluminaIterativeEaterIterator implements Iterator<IlluminaSpot>, DataEater<IlluminaSpot, Object>
+IlluminaIterativeEaterIterator implements Iterator<IlluminaSpot>, DataConsumer<IlluminaSpot, Object>

```

Comment:
Remove IlluminaIterativeEaterIterator. I think this is only used in IlluminaIterativeEaterIteratorTest. Also remove IlluminaIterativeEaterIteratorTest.

Reply:
This class is being used inside IlluminaIterativeConsumer which itself is being used by FastqScanner. FastqScanner on the other hand is in use by ReadsValidator which, im quite certain, I have seen being used inside another repository. So i dont think its possible to remove this class completely.